src/m365/spo/commands/page/page-copy.ts

```diff
@@ -0,0 +1,163 @@
+    }
+    if (targetFullName.startsWith('/')) {
+      targetFullName = targetFullName.substring(1);
+    }
+
+    if (this.verbose) {
+      logger.logToStderr(`Creating page copy...`);
+    }
+
+    const requestOptions: any = {
+      url: `${args.options.webUrl}/_api/SP.MoveCopyUtil.CopyFileByPath()`,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      data: {
+        srcPath: { DecodedUrl: `${args.options.webUrl}/sitepages/${sourceFullName}` },
+        destPath: { DecodedUrl: `${targetSiteUrl}/sitepages/${targetFullName}` },
+        options: { ResetAuthorAndCreatedOnCopy: true, ShouldBypassSharedLocks: true },
+        overwrite: !!args.options.overwrite
+      },
+      responseType: 'json'
+    };
+
+    request
+      .post<void>(requestOptions)
+      .catch((err: any): void => {
+        // The API returns a 400 when file already exists
+        if (err && err.response && err.response.status && err.response.status === 400) {
+          return;
+        }
+        // Throw the error if something else happened
+        throw err;
+      })
+      .then((): Promise<ClientSidePageProperties> => {
+        const requestOptions: any = {
+          url: `${targetSiteUrl}/_api/sitepages/pages/GetByUrl('sitepages/${targetFullName}')`,
+          headers: {
+            'accept': 'application/json;odata=nometadata'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<ClientSidePageProperties>(requestOptions)
+      })
+      .then((res: ClientSidePageProperties): void => {
+        logger.log(res);
+
+        if (this.verbose) {
+          logger.logToStderr(chalk.green('DONE'));
+        }
+
+        cb();
+      })
+      .catch((err: any): void => {
+        return this.handleRejectedODataJsonPromise(err, logger, cb)
+      });
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '--sourceName <sourceName>',
+        description: 'The name of the source file'
+      },
+      {
+        option: '--targetUrl <targetUrl>',
+        description: 'The url of the target file. You are able to provide the page its name, relative path, or absolute path'
+      },
+      {
+        option: '-u, --webUrl <webUrl>',
+        description: 'URL of the site where the page should be created'
+      },
+      {
+        option: '--overwrite',
+        description: 'Overwrite the target page when it already exists'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    const isValidSharePointUrl: boolean | string = SpoCommand.isValidSharePointUrl(args.options.webUrl);
+    if (isValidSharePointUrl !== true) {
+      return isValidSharePointUrl;
+    }
+
+    return true;
+  }
+
+  private getSiteUrl(webUrl: string, targetFullName: string): { siteUrl: string, pageName: string } {
```

Comment:
Let's rename this method to `getTargetSiteUrl` to clearly denote that it's used for calculating the destination URL.